mongoengine/base/metaclasses.py

```diff
@@ -391,10 +391,16 @@ def __new__(cls, name, bases, attrs):
             new_class._fields['id'] = ObjectIdField(db_field='_id')
```

Comment:
I think the whole problem for the failing test is here. We are using auto_id to always force it to 'id'. But when there is already for example a hearted field called 'id' which is not a primary_key, we are creating a huge mess.

I see two possible solutions:
1) raise an error, but breaking change and no big advantage
2) Instead of using a fixed 'id', use a call to an 'id' generator, that will compose something containing 'id', paying attention to what else is in the document